os/fs/smartfs/smartfs_smart.c

```diff
@@ -1922,45 +1950,60 @@ int smartfs_rename(struct inode *mountpt, const char *oldrelpath, const char *ne
 		mode = oldentry.flags & SMARTFS_DIRENT_MODE;
 		type = oldentry.flags & SMARTFS_DIRENT_TYPE;
 
-		ret = smartfs_createentry(fs, newparentdirsector, newfilename, type, mode, &newentry, oldentry.firstsector, NULL);
-		if (ret != OK) {
-			goto errout_with_semaphore;
-		}
+#ifdef CONFIG_SMARTFS_USE_SECTOR_BUFFER
+		if (oldparentdirsector == newparentdirsector) {
+			ret = smartfs_replaceentry(fs, oldentry, newfilename, type, &newentry, NULL);
+			if (ret != OK) {
+				goto errout_with_semaphore;
+			}
+		} else if (smartfs_findinvalidentry(fs, newparentdirsector, &invalidentry)) {
+			ret = smartfs_replaceentry(fs, invalidentry, newfilename, type, &newentry, NULL);
+			if (ret != OK) {
+				goto errout_with_semaphore;
+			}
+		} else
+#endif
+		{
+			ret = smartfs_createentry(fs, newparentdirsector, newfilename, type, mode, &newentry, oldentry.firstsector, NULL);
```

Comment:
inside of smartfs_createentry, there's code under CONFIG_SMARTFS_USE_SECTOR_BUFFER.
It is useless after we decide use replaceentry instead of createentry.
Please clean it. (At least put TODO comment there)

Reply:
Yes, agreed, thanks.